### ISSUE TITLE
How to load multi-label data  for classification??

### ISSUE THREAD

**Issue by nitish11** [fchollet/keras#747]
Hi,

My application is similar to cifar100. I am working on multi-label classification and I have many such labels in an image. 
For example, lets say I have an image with two cars and a ship.  I want to get an output that the image has a car and ship.Also, I want to classify it saying that it has two cars and a ship.

To explain it further, I will give the sample of my label file.

Lets suppose my images have car, bike, truck, aeroplane and ships.
In multi-label classification, my label file would be of the form:
image_name [-1 1 -1 1 1] {1 is class present, -1 is class absent}
I am making an assumption that each image has atmost 3 components. So my label file for an image with 2 cars and ship would be:
image_name [[-1 1 -1 -1 -1], [-1 1 -1 -1 -1],[-1 -1 -1 1 -1]]

How to load these data in keras ?? Is there any example code to upload the labels and the images separately???

**Comment by fchollet** [comment]
One way to do it
- use label vectors that are mostly zeros, with ones at the index of each class present in the input (default assumption is 0 == class is absent).
- use `binary_crossentropy` as loss
- compute accuracy metrics yourself via the output of `model.predict`

Another way: 
- use label vectors that are 1s and -1s like your describe
- write your own loss function to handle them

**Comment by nitish11** [comment]
Thanks @fchollet .

I want to go with the first approach. <b>  Can you suggest me some sample code/ example to load multi-label labels and data ?? </b>

In reference to the above example, if I have a folder containing all the images and making an assumption that each image has atmost 3 components. So my label file for an image with 2 cars and ship would be:
image_name [[0 1 0 0 0], [0 1 0 0 0],[0 0 0 1 0]]

**Comment by hemanth2090** [comment]
@fchollet @nitish11  Were you able to solve the above question? I have a similar application where I have multiple labels for each image. I have also stored the label as a vector of vectors.

**Comment by alyato** [comment]
@nitish11 Do you share your code how to load  load multi-label data.
